libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -231,29 +234,88 @@ export function getChecklistObjectFromHdf(
   return _.get(hdf, 'passthrough.checklist', EmptyChecklistObject);
 }
 
-export function validateChecklistMetadata(
-  metadata: Pick<Asset, 'hostip' | 'hostfqdn' | 'hostmac'>
-): {isError: boolean; message: string; invalid: string[]} {
-  const errors: string[] = [];
-  const invalid: string[] = [];
-
-  // STIG Viewer can autofill the FQDN as the local IP address
-  if (
-    metadata.hostfqdn &&
-    !(isFQDN(metadata.hostfqdn) || isIP(metadata.hostfqdn))
-  ) {
-    errors.push(`Invalid host FQDN: ${metadata.hostfqdn}`);
-    invalid.push('hostfqdn');
-  }
-  if (metadata.hostip && !isIP(metadata.hostip)) {
-    errors.push(`Invalid host IP address: ${metadata.hostip}`);
-    invalid.push('hostip');
+const checklistMetadataSchema: Revalidator.JSONSchema<Asset> = {
+  properties: {
+    hostfqdn: {
+      type: 'string',
+      // STIG Viewer can autofill the FQDN as the local IP address
+      conform: (fqdn: string) => !fqdn || isFQDN(fqdn) || isIP(fqdn),
+      message: 'Host FQDN'
+    },
+    hostip: {
+      type: 'string',
+      conform: (ip: string) => !ip || isIP(ip),
+      message: 'Host IP'
+    },
+    hostmac: {
+      type: 'string',
+      conform: (mac: string) => !mac || isMACAddress(mac),
+      message: 'Host MAC'
+    },
+    role: {
+      type: 'string',
+      enum: Object.values(Role),
+      message: 'Role'
+    },
+    assettype: {
+      type: 'string',
+      enum: Object.values(Assettype),
+      message: 'Asset Type'
+    },
+    techarea: {
+      type: 'string',
+      enum: Object.values(Techarea),
+      message: 'Tech Area'
+    },
+    webordatabase: {
+      type: 'boolean',
+      message: 'Web or Database STIG'
+    }
   }
-  if (metadata.hostmac && !isMACAddress(metadata.hostmac)) {
-    errors.push(`Invalid host MAC address: ${metadata.hostmac}`);
-    invalid.push('hostmac');
+};
+
+const profileMetadataSchema: Revalidator.JSONSchema<StigMetadata> = {
+  properties: {
+    version: {
+      type: 'integer',
+      minimum: 0,
+      message: 'Version must be a positive integer'
+    },
+    releasenumber: {
+      type: 'integer',
+      minimum: 0,
+      message: 'Release number must be a positive integer'
+    },
+    releasedate: {
+      type: 'string',
+      conform: (date: string) => !date || isDate(date),
+      message: 'Release date must be a valid date'
+    }
   }
-  return {isError: errors.length > 0, message: errors.join('\n'), invalid};
+};
+
+export function validateChecklistMetadata(
```

Comment:
after discussion, we are going to move forward with having a helper function exposed in the api to do the checklist metadata validation.  this function and the contructors will utilize our internal functions for validating the asset information and the profile information.

Reply:
Done. Validation functions are both still there but can be called both at once from a new single checklist metadata validation method. This makes the logic on the frontend much simpler.